### ISSUE TITLE
Permission request for location should popup after signin

### ISSUE THREAD

**Issue by charliemangano** [PeriodPals/periodpals#78]
# Permission request for location should popup after SignIn
The permission request pop-up for the user's location displays when opening the app for the first time.

**To Reproduce**
Install (or delete cache if already installed) and launch the app.

**Expected behavior**
The pop-up should display only after the user has logged in/signed up succesfully

**Screenshots**
![image](https://github.com/user-attachments/assets/107d1699-2110-43d4-9e16-e8f478e113e3)

**Additional context**
We need this fixed in order to do the end-to-end tests (#48)

**Comment by lazarinibruno** [comment]
The development page for android states: 

_When a feature in your app needs location access, wait until the user interacts with the feature before making the permission request. _

Therefore, we are indeed asking for user permission in the wrong moment. I think we should ask for permission when the user clicks on the **map, alert list or alert**.

**Comment by charliemangano** [comment]
Duplicate of #95. Leaving both issues open as there are useful information on both.
